packages/@ama-sdk/core/src/plugins/mgw-mdw-auth/mgw-mdw-auth.request.ts

```diff
@@ -1,22 +1,45 @@
 import { v4 } from 'uuid';
-import { ab2str, str2ab } from '../../utils/crypto';
-import { createBase64Encoder, createBase64UrlEncoder } from '../../utils/json-token';
+import { base64EncodeUrl, createBase64Encoder, createBase64UrlEncoder } from '../../utils/json-token';
 import { PluginRunner, RequestOptions, RequestPlugin } from '../core';
+import type { createHash as createHashType, createHmac as createHmacType } from 'node:crypto';
+
+
+/**
+ *
+ * @param value
+ */
+export async function sha256Browser(value: string) {
+  const utf8 = new TextEncoder().encode(value);
+  const hashBuffer = await crypto.subtle.digest('SHA-256', utf8);
+  const hashArray = Array.from(new Uint8Array(hashBuffer));
+  const hashHex = hashArray
+    .map((bytes) => bytes.toString(16).padStart(2, '0'))
+    .join('');
+  return hashHex;
+}
+
+/**
+ *
+ * @param value
+ */
+export function sha256Node(value: string) {
+  const { createHash }: { createHash: typeof createHashType } = require('node:crypto');
+  return createHash('sha256')
+    .update(value, 'latin1')
+    .digest('hex');
+}
 
 /**
  * Computes the SHA256 digest of the given string
  * @param value Value to hash
  */
 export async function sha256(value: string) {
-  // Web browsers
   if (typeof window !== 'undefined' && typeof window.crypto !== 'undefined') {
-    return ab2str(await crypto.subtle.digest('SHA-256', str2ab(value)));
+    return sha256Browser(value);
```

Comment:
Depending on which node.js versions you target, maybe you do not really need two different versions for node.js and the browser. It looks like the browser version also works in node.js (v20.11.0).

Reply:
indeed, seems the case from Node 15 (cf. https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest)